### PR TITLE
Add String() for enums

### PR DESCRIPTION
**Pull request by gabriel**
Allows you to do someEnum.String() to get string value instead of some awkward Rev map lookup

### CODE REVIEW

**Comment by gabriel** [comment]
@maxtaco

**Comment by gabriel** [comment]
Is there a reason enum strings should be all caps? We're not in C.

**Comment by mlsteele** [comment]
Last time I or song tried to add this, it turned out that a bunch of generated types already have custom `String` methods and that they're not all of the same form. And it's not clear what to do with invalid values.

**Comment by mlsteele** [comment]
For example https://github.com/keybase/client/blob/miles/CORE-4177-udcache/go/protocol/keybase1/extras.go#L572

But if you have tried compiling the client with this PR then I'm confusing this with something else.

**Comment by gabriel** [comment]
The existing enums in keybase1 are all used for descriptions, so we can nuke them without breaking anything.
The existing enums in chat1 are: MessageType, TopicType and their values are same as this default, so can be removed, and can nuke them without breaking anything.

**Comment by gabriel** [comment]
https://play.golang.org/p/R4zLEXqusZ

**Comment by gabriel** [comment]
For example, this would be equivalent of existing chat1 String() method:
https://github.com/keybase/client/blob/master/go/protocol/chat1/extras.go#L68

**Comment by mlsteele** [comment]
Alright. Thanks for the example! Figuring out where the existing String methods are used (to verify that the change in behavior won't break stuff) seems pretty tough given that they could be in `%s` and `%v` subs. Just wanted to make sure you had those tricky things in mind :)
I defer to @maxtaco

**Comment by maxtaco** [comment]
Is this a risky change? I really hate to take risks if we're not making the user experience better.

**Comment by maxtaco** [comment]
@gabriel --- @mlsteele pointed to an example in which the auto-generated value would be different from what's already there....  What would the story for that be?

**Comment by gabriel** [comment]
ok.. I added a `@go("nostring")` tag which will skip generation of the method

This means we can whitelist those enums with custom `String()` methods.

**Comment by gabriel** [comment]
Here is the PR for go nostring directives for the protocol for the enums that have `String()`:
https://github.com/keybase/client/pull/5096

File: src/emit.iced [hunk 1]
```diff
@@ -283,11 +283,12 @@ exports.GoEmitter = class GoEmitter
       when "fixed"
         @emit_fixed type
       when "enum"
-        @emit_enum type
+        nostring = (type.go == "nostring")
```
Comment (maxtaco) [inline src/emit.iced]: Use `is`

File: src/emit.iced [hunk 2]
```diff
@@ -320,6 +321,18 @@ exports.GoEmitter = class GoEmitter
     @untab()
     @output "}"
 
+    if !nostring
```
Comment (maxtaco) [inline src/emit.iced]: `unless nostring`

**Comment by gabriel** [comment]
@maxtaco I guess we should push the new version? Not sure how to do that... once that's done I could update the protocol PR

**Comment by maxtaco** [comment]
ok done!

**Comment by gabriel** [comment]
Oops had to add a fix here:
https://github.com/keybase/node-avdl-compiler/pull/11

**Pull request merged**